compiler/src/dotty/tools/dotc/transform/Erasure.scala

```diff
@@ -626,17 +626,19 @@ object Erasure extends TypeTestsCasts{
 
                     if (isRequired) {
                       // check for clashes
-                      val clash: Option[Symbol] = oldSymbol.owner.info.decls.lookupAll(bridge.name).find {
-                        sym =>
-                          (sym.name eq bridge.name) && sym.info.widen =:= bridge.info.widen
-                      }.orElse(
-                        emittedBridges.find(stat => (stat.name == bridge.name) && stat.tpe.widen =:= bridge.info.widen)
+                      val clash: Option[Symbol] = oldSymbol.owner.info.memberExcluding(bridge.name, Flags.Bridge).alternatives.find {
+                        denot => {
+                          !denot.symbol.owner.is(Flags.Trait) && (denot.symbol.owner ne defn.ObjectClass) &&
```

Comment:
> !denot.symbol.owner.is(Flags.Trait)

why is clashing with trait methods fine?

> denot.symbol.owner ne defn.ObjectClass

Why?

> !denot.symbol.owner.derivesFrom(bridge.owner) 

Is  `denot.symbol.owner` ever a non-trivial subclass of `bridge.owner` ?
If yes, would be nice to add a test.